lib/SILOptimizer/Mandatory/TFLowerGraph.cpp

```diff
@@ -1724,6 +1725,16 @@ GLStatus TFGraphLowering::visitGraphOperationInst(GraphOperationInst *inst) {
     }
   }
 
+  // This helper function decodes a shape attribute into its values.
+  auto decodeShapeAttr =
+    [&](SymbolicValue attr, SmallVectorImpl<int64_t> &result) {
+    attr = attr.lookThroughSingleElementAggregates();
+    for (auto elt : attr.getArrayValue()) {
```

Comment:
To confirm: we are guaranteed to generate valid SIL as input to graph lowering, so that here we won't trigger this kind of assert: `assert(getKind() == Array)`, correct?

Reply:
Correct.  The invariant we need to preserver is that all graph_op instructions are valid.  tf deabstraction should verify this (in the code referenced above) and any other special cases forming graph ops should always be assumed to be correct.

Reply:
The rationale here is that we don't *want* this code to be in the business of generating diagnostics.  It is better to generate them earlier.